phpunit/html/wp-html-walker-test.php

```diff
@@ -0,0 +1,1012 @@
+		$this->assertFalse( $w->next_tag( 'p' ) );
+		$this->assertNull( $w->get_attribute( 'class' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_get_attribute_returns_null_when_attribute_missing() {
+		$w = new WP_HTML_Walker( '<div class="test">Test</div>' );
+		$this->assertTrue( $w->next_tag( 'div' ) );
+		$this->assertNull( $w->get_attribute( 'test-id' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_get_attribute_returns_attribute_value() {
+		$w = new WP_HTML_Walker( '<div class="test">Test</div>' );
+		$this->assertTrue( $w->next_tag( 'div' ) );
+		$this->assertSame( 'test', $w->get_attribute( 'class' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_get_attribute_returns_true_for_boolean_attribute() {
+		$w = new WP_HTML_Walker( '<div enabled class="test">Test</div>' );
+		$this->assertTrue( $w->next_tag( array( 'class_name' => 'test' ) ) );
+		$this->assertTrue( $w->get_attribute( 'enabled' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_calling_tostring_applies_the_updates_so_far_and_keeps_the_walker_on_the_current_tag() {
+		$w = new WP_HTML_Walker( '<hr id="remove" /><div enabled class="test">Test</div><span id="span-id"></span>' );
+		$w->next_tag();
+		$w->remove_attribute( 'id' );
+
+		$w->next_tag();
+		$w->set_attribute( 'id', 'div-id-1' );
+		$w->add_class( 'new_class_1' );
+		$this->assertSame( '<hr  /><div id="div-id-1" enabled class="test new_class_1">Test</div><span id="span-id"></span>', (string) $w );
+
+		$w->set_attribute( 'id', 'div-id-2' );
+		$w->add_class( 'new_class_2' );
+		$this->assertSame( '<hr  /><div id="div-id-2" enabled class="test new_class_1 new_class_2">Test</div><span id="span-id"></span>', (string) $w );
+
+		$w->next_tag();
+		$w->remove_attribute( 'id' );
+		$this->assertSame( '<hr  /><div id="div-id-2" enabled class="test new_class_1 new_class_2">Test</div><span ></span>', (string) $w );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_get_attribute_returns_string_for_truthy_attributes() {
+		$w = new WP_HTML_Walker( '<div enabled=enabled checked=1 hidden="true" class="test">Test</div>' );
+		$this->assertTrue( $w->next_tag( array() ) );
+		$this->assertSame( 'enabled', $w->get_attribute( 'enabled' ) );
+		$this->assertSame( '1', $w->get_attribute( 'checked' ) );
+		$this->assertSame( 'true', $w->get_attribute( 'hidden' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_treats_slash_as_attribute_separator() {
+		$w = new WP_HTML_Walker( '<div a/b/c/d/e="test">Test</div>' );
+		$this->assertTrue( $w->next_tag( array() ) );
+		$this->assertTrue( $w->get_attribute( 'a' ) );
+		$this->assertTrue( $w->get_attribute( 'b' ) );
+		$this->assertTrue( $w->get_attribute( 'c' ) );
+		$this->assertTrue( $w->get_attribute( 'd' ) );
+		$this->assertSame( 'test', $w->get_attribute( 'e' ) );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_to_string_with_no_updates_returns_the_original_html() {
```

Comment:
```suggestion
	public function test_tostring_with_no_updates_returns_the_original_html() {
```
An earlier test uses `tostring` rather than `to_string`. Let's keep this consistent.